patches/openwrt/0007-xrx200-ensure-the-correct-version-for-fritz7360.patch

```diff
@@ -0,0 +1,94 @@
+new file mode 100644
+index 0000000000..f0520502ce
+--- /dev/null
++++ b/target/linux/lantiq/xrx200/base-files/lib/preinit/01_sysinfo.sh
+@@ -0,0 +1,75 @@
++set_sysinfo_xrx200() {
++    local board_name=$1
++    local model=$2
++
++    echo "${board_name}" > /tmp/sysinfo/board_name
++    echo "${model}" > /tmp/sysinfo/model
++}
++
++do_sysinfo_xrx200_for_fritz7360_model() {
++    local board_name=$1
++    local model=$2
++    local urlader_version urlader_memsize urlader_flashsize
++    local hexdump_format='4/1 "%02x""\n"'
++
++    # Values are based on urlader-parser-py
++    # https://github.com/grische/urlader-parser-py/blob/42970bf8dec7962317df4ff734c57ebf36df8905/parser.py#L77-L84
++    urlader_version="$(dd if=/dev/mtd0ro bs=1 skip=$((0x580+0x0)) count=4 | hexdump -e "${hexdump_format}")"
++    if [ "${urlader_version}" != "00000003" ]; then
++        >&2 echo "Warning: unexpected urlader version found: ${urlader_version}"
++        set_sysinfo_xrx200 "${board_name}" "${model}"
++        return
++    fi
++
++    urlader_memsize="$(dd if=/dev/mtd0ro bs=1 skip=$((0x580+0x4)) count=4 | hexdump -e "${hexdump_format}")"
++    if [ "${urlader_memsize}" != "08000000" ]; then
++        >&2 echo "Warning: unexpected memsize found: ${urlader_memsize}"
++        set_sysinfo_xrx200 "${board_name}" "${model}"
++        return
++    fi
++
++    urlader_flashsize="$(dd if=/dev/mtd0ro bs=1 skip=$((0x580+0x8)) count=4 | hexdump -e "${hexdump_format}")"
++     case "${urlader_flashsize}" in
++        "02000000")    # 32MB
++            # see vr9_avm_fritz7360-v2.dts
++            board_name="avm,fritz7360-v2"
++            model="AVM FRITZ!Box 7360 V2"
++            ;;
++        "01000000")    # 16MB
++            # see vr9_avm_fritz7360sl.dts
++            board_name="avm,fritz7360sl"
++            model="AVM FRITZ!Box 7360 SL"
++            ;;
++        *)
++            >&2 echo "Warning: unexpected flashsize found: ${urlader_flashsize}"
++            ;;
++    esac
++
++    set_sysinfo_xrx200 "${board_name}" "${model}"
++}
++
++do_sysinfo_xrx200() {
++    local reported_board board_name model
++
++    [ -d /proc/device-tree ] || return
++    reported_board="$(strings /proc/device-tree/compatible | head -1)"
++    reported_model="$(cat /proc/device-tree/model)"
++
++    mkdir -p /tmp/sysinfo
++    # 7360 is notoriously known for not writing "v2" on their labels and many
++    # routers have flashed the wrong firmware with the wrong flash layout.
++    # We ensure that the underlying hardware is reported correctly, so that
++    # future upgrades will use the correct flash layout.
++    # Using 7360v2 hardware, an upgrade from a 7360v1/sl firmware to a 7360v2
++    # is working.
++    case "${reported_board}" in
++        *7360*)
++            do_sysinfo_xrx200_for_fritz7360_model "${reported_board}" "${reported_model}"
++            ;;
++        *)
++            set_sysinfo_xrx200 "${reported_board}" "${reported_model}"
```

Comment:
Is this necessary? Read, wouldn't a simple `return` achieve the same result?

Reply:
I wanted to explicitly set it ourselves, to be independent of potential OpenWRT changes.

If you think that falling back to the [upstream OpenWRT detection](https://github.com/openwrt/openwrt/blob/openwrt-22.03/package/base-files/files/lib/preinit/02_sysinfo) is a better, fine with me.

Let me know what you prefer.

Reply:
Please use the upstream one

Reply:
Done